src/kiwixapp.cpp

```diff
@@ -507,18 +508,40 @@ void KiwixApp::postInit() {
 
 void KiwixApp::handleItemsState(TabType tabType)
 {
-    auto libraryOrSettingsTab =  (tabType == TabType::LibraryTab || tabType == TabType::SettingsTab);
+    auto libraryOrSettingsTab = (tabType == TabType::LibraryTab || tabType == TabType::SettingsTab);
     auto notBookmarkableTab = libraryOrSettingsTab || getTabWidget()->currentArticleUrl().isEmpty();
+    auto hasZimFile = !getTabWidget()->currentZimId().isEmpty();
     auto app = KiwixApp::instance();
+
+    // Navigation actions
     app->getAction(KiwixApp::ToggleTOCAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::OpenHomePageAction)->setDisabled(!hasZimFile);
+    app->getAction(KiwixApp::RandomArticleAction)->setDisabled(!hasZimFile);
+
+    // Reading & bookmarks
     app->getAction(KiwixApp::ToggleReadingListAction)->setDisabled(libraryOrSettingsTab);
     app->getAction(KiwixApp::ToggleAddBookmarkAction)->setDisabled(notBookmarkableTab);
+    app->getAction(KiwixApp::ReadArticleAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::ReadTextAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::ReadStopAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::ToggleTTSLanguageAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::ToggleTTSVoiceAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::IncreaseTTSSpeedAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::DecreaseTTSSpeedAction)->setDisabled(libraryOrSettingsTab);
+
+    // Search & zoom
     app->getAction(KiwixApp::FindInPageAction)->setDisabled(libraryOrSettingsTab);
+    app->getAction(KiwixApp::SearchArticleAction)->setDisabled(libraryOrSettingsTab);
```

Comment:
Multi-zim search options (hiding under the misleadingly named `OpenMultiZimAction` action) make it possible searching in any ZIM file while on any tab. It can be a confusing experience on the library and/or settings tab (because of the default selection of the ZIM file to search in) but we shouldn't rather fix it instead of disabling that functionality.